src/utils/mobileOSDetectAsync.utils.ts

```diff
@@ -0,0 +1,93 @@
+import UAParser from "ua-parser-js";
+import { huaweiDevicesRegex, validCodes } from "../constants/mobile-devices.constants";
+
+/**
+ * This file contains utility functions and types for detecting the mobile operating system.
+ * It uses the User-Agent string and the User-Agent Client Hints API to determine the OS.
+ */
+
+declare global {
+    interface Window {
+        MSStream?: {
+            msClose: () => void;
+            msDetachStream: () => void;
+            readonly type: string;
+        };
+        opera?: string;
+    }
+    interface Navigator {
+        userAgentData?: NavigatorUAData;
```

Comment:
```suggestion
        userAgentData?: NavigatorUAData;
```
This is a browser global environment default. You should not need to type this. If it is not already typed, you should check with the `tsconfig.json` to make sure the proper environment is set

Reply:
in the browser global environment we are only getting `userAgent` (which is a string)

`userAgentData` has more data that is needed to differentiate certain Huawei devices from other Android devices

Not sure how to "properly set the environment" in `tsconfig.json`